logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/auth/LogoutUseCase.kt

```diff
@@ -70,38 +72,55 @@ internal class LogoutUseCaseImpl @Suppress("LongParameterList") constructor(
             userSessionWorkScheduler.cancelScheduledSendingOfPendingMessages()
 
             when (reason) {
-                LogoutReason.SELF_HARD_LOGOUT -> {
-                    // we put this delay here to avoid a race condition when
-                    // receiving web socket events at the exact time of logging put
-                    delay(CLEAR_DATA_DELAY)
-                    clearClientDataUseCase()
-                    clearUserDataUseCase() // this clears also current client id
-                }
-
+                LogoutReason.SELF_HARD_LOGOUT -> wipeAllData()
                 LogoutReason.REMOVED_CLIENT, LogoutReason.DELETED_ACCOUNT -> {
-                    // receiving web socket events at the exact time of logging put
-                    clearClientDataUseCase()
-                    logoutRepository.clearClientRelatedLocalMetadata()
-                    clientRepository.clearCurrentClientId()
-                    clientRepository.clearHasRegisteredMLSClient()
-                    // After logout we need to mark the Firebase token as invalid
-                    // locally so that we can register a new one on the next login.
-                    pushTokenRepository.setUpdateFirebaseTokenFlag(true)
+                    if (kaliumConfigs.wipeOnDeviceRemoval) {
+                        wipeAllData()
+                    } else {
+                        wipeTokenAndMetadata()
+                    }
                 }
-
-                LogoutReason.SELF_SOFT_LOGOUT, LogoutReason.SESSION_EXPIRED -> {
-                    clientRepository.clearCurrentClientId()
-                    // After logout we need to mark the Firebase token as invalid
-                    // locally so that we can register a new one on the next login.
-                    pushTokenRepository.setUpdateFirebaseTokenFlag(true)
+                LogoutReason.SESSION_EXPIRED -> {
```

Comment:
since this case has the same code as `REMOVED_CLIENT` and `DELETED_ACCOUNT`, can we have them all in one place?

Reply:
It doesn't have the same code